img/deconv_foxsi_simple.pro

```diff
@@ -1,205 +1,141 @@
+;	ITER	An array of iterations for which to return the deconvolved map.
+;			Default: [1,2,3,4,5,10,20,50]
+;	RECONV_map	The reconvolved map (deconvolved convolved with PSF) for each iteration
+;	CSTAT	Cash statistic for each iteration
+;
+; HISTORY:
+;		2020-feb-12		LG	Updated routines to make it easier to use, added PSF input.
+;		2016-oct-30		LG	Created from scraps of my analysis code.
+;----
+
+FUNCTION DECONV_FOXSI_SIMPLE, MAP, FOV=FOV, PSF_IN=PSF_IN, iter=iter, $
+					   RECONV_map = reconv_map, CSTAT=cstat, STOP = STOP
+					   					   
 	default, fov, 4.
-	default, erange, [5.,12.]
-	default, psf_smooth, 15		; only applies to measured PSF
-	default, img_smooth, 15
-	default, rotation, 79
-	default, iter, [1,2,3,4,5,10,20,40,80,100]
+	default, iter, [1,2,3,4,5,10,20,50]
 		
-	dim = fov*60./pix
-
-	loadct, 5
-
-
-
-	; Step 1: Define a PSF map normalized to unity
-	; Use either the X5 measured values for 7' off-axis or else the 3-component 
-	; 2D Gaussian PSF that came from a fit to the measurement.
+	; Step 1: Prepare the PSF map.
+	; If a PSF was included as a keyword, use that one.
+	; If not, restore the standard on-axis one.
+	; Normalize the PSF to unity
 	
-	print
-	print, 'Defining PSF...'
-	print
+	if keyword_set (PSF_IN) then psf = psf_in else $
+		restore, GETENV('FOXSIDB') + '/../calibration_data/foxsi_psf_on_axis.sav'
 	
-		; or, use a gaussian PSF
-		params = [ 0.0, 0.0, 0.0, 0.9875, 0.218387, 0.0762158, 1.27836, 1.77492, 4.36214,$
-		   7.21397, 47.5, 240.314, 0.0 ]	; parameters from Steven
-		   	
-		psf1 = psf_gaussian( npix=[dim,dim], /double, st_dev=[1.27836, 1.77492]/pix )
-		psf2 = psf_gaussian( npix=[dim,dim], /double, $
-							 st_dev=2.*[4.36214, 7.21397]/pix )*params[4]
-		psf3 = psf_gaussian( npix=[dim,dim], /double, $
-							 st_dev=2.*[47.5, 240.314]/pix )*params[4]*params[5]
-		psftest = make_map( psf1+psf2+psf3, dx=pix, dy=pix )
-		psf = rot_map( psftest, -45 )
 		psf.roll_angle=0.
- 
+ 		psf = make_submap( PSF, cen=[psf.xc,psf.yc], fov=fov )
  		psf.data = psf.data / total(psf.data)	; Renormalize
- 		size = size(psf.data)
+ 		;;;size = size(psf.data)
  		
- 		; rebin to detector pixel size.
- 		;new_data = frebin( psf.data, size[1]/7.735, size[2]/7.735, /total )
- 		;psf = make_map( new_data, xcen=psf.xc, ycen=psf.yc, dx=7.735, dy=7.735 )
-
-
 	;
  	; Step 2: Process measured image
  	;
  
-	print
- 	print, 'Setting up raw image...'
-	print
- 
-;	imdim = 1000/fix(pix)
-	imdim=128*7.78/pix
-
-;	if keyword_set(stop) then stop
-
-	; Find and isolate a FOV around the source.
-		first = map
-		first_cen = map_centroid( first, thr=0.1*max(first.data) )
-		first = make_submap( first, cen=first_cen, fov=3 )
-		; this gets returned in keyword FIRST to use as sample unprocessed image.
+	; Find the source and choose a FOV centered on it.
+	; This is so that you don't have to deconvolve a large image.
+	map_cen = map_centroid( map, thr=0.1*max(map.data) )
```

Comment:
When I was using this code, I found that sometimes this part of the code was not doing a great job at centering the FOXSI-2 microflares, depending on the other emission observed in the FOV. In my own version, I added the keyword "map_cen" and changed line 76 to:

if keyword_set(map_cen) then map_cen = map_cen else map_cen = map_centroid( map, thr=0.1*max(map.data) )

This allows users to set the map center if the centroid of the object of interest is already known. 

The code functions without this change, so this pull request could be merged and I could submit an additional pull request for these changes, if that is preferred.